city_metrix/layers/layer.py

```diff
@@ -325,6 +325,18 @@ def get_stats_funcs(stats_func):
         return [stats_func]
 
 
+def set_resampling_method(image: ee.Image, resampling_method: str):
+    valid_raster_resampling_methods = ['bilinear', 'bicubic']
+
+    if resampling_method not in valid_raster_resampling_methods:
+        raise ValueError(f"Invalid resampling method ('{resampling_method}'). "
+                         f"Valid methods: {valid_raster_resampling_methods}")
+
+    data = image.resample(resampling_method)
```

Comment:
I'm not sure how this would play into, but in `get_image_collection`, you might accidentally trigger some sort of resampling by virtue of reading a raster at a given `scale` (which may be different than the native resolution of the raster). Do you need to make sure you're resampling correctly in the Xee read, or does it seem to handle it properly by calling this function here?

Reply:
Resampling needs to be specified on the images before calling XEE as far as I've found with my investigations.